packages/api/src/external/carequality/document/process-document-retrieval-results.ts

```diff
@@ -0,0 +1,170 @@
+import { out } from "@metriport/core/util/log";
+import { Bundle, BundleEntry } from "@medplum/fhirtypes";
+import { errorToString } from "@metriport/core/util/error/shared";
+import { capture } from "@metriport/core/util/notifications";
+import { DocumentReference } from "@metriport/ihe-gateway-sdk";
+import { MedicalDataSource } from "@metriport/core/external/index";
+import { DocumentRetrievalResult } from "../document-retrieval-result";
+import { isConvertible } from "../../fhir-converter/converter";
+import { setDocQueryProgressWithSource } from "../../hie/set-doc-query-progress-with-source";
+import { tallyDocQueryProgressWithSource } from "../../hie/tally-doc-query-progress-with-source";
+import { convertCDAToFHIR } from "../../fhir-converter/converter";
+import { upsertDocumentsToFHIRServer } from "../../fhir/document/save-document-reference";
+import { cqToFHIR } from "../../fhir/document";
+import { getDocuments } from "../../fhir/document/get-documents";
+import { ingestIntoSearchEngine } from "../../aws/opensearch";
+
+export async function processDocumentRetrievalResults({
+  requestId,
+  patientId,
+  cxId,
+  documentRetrievalResults,
+}: {
+  requestId: string;
+  patientId: string;
+  cxId: string;
+  documentRetrievalResults: DocumentRetrievalResult[];
+}): Promise<void> {
+  try {
+    for (const result of documentRetrievalResults) {
```

Comment:
It took me some time to understand `result` means the result of a single external GW. What do you think we name `results` as `resultsOfExternalGW` and `documentRetrievalResults` as `resultsOfAllExternalGWs`?